src/parse.rs

```diff
@@ -102,31 +95,59 @@ pub enum ParseResult {
     Unsupported(SentenceType),
 }
 
+#[derive(Debug, Clone, PartialEq)]
+pub enum NmeaError<'a> {
+    /// The provided input was not a proper UTF-8 string
+    Utf8DecodingError,
+    /// The checksum of the sentence was corrupt or wrong
+    ChecksumMismatch,
+    /// For some reason a sentence was passed to the wrong sentence specific parser, this error
+    /// should never happen. First slice is the expected header, second is the found one
+    WrongSentenceHeader(&'a [u8], &'a [u8]),
+    /// The sentence could not be parsed because its format was invalid
+    ParsingError(nom::Err<(&'a [u8], nom::error::ErrorKind)>),
+    /// The sentence was too long to be parsed, our current limit is 102 characters
+    SentenceLength(usize),
+    /// The type of a GSV sentence was not a valid Gnss type
+    InvalidGnssType,
+    /// The sentence has and maybe will never be implemented
+    Unsupported(SentenceType),
+    /// The provided navigation configuration was empty and thus invalid
+    EmptyNavConfig,
+}
+
+impl<'a> From<nom::Err<(&'a [u8], nom::error::ErrorKind)>> for NmeaError<'a> {
+    fn from(error: nom::Err<(&'a [u8], nom::error::ErrorKind)>) -> Self {
+        Self::ParsingError(error)
+    }
+}
+
```

Comment:
May be implement std::error::Error for our Error type?

Reply:
Since the final target for this is to be no_std anyways I'm not quite sure whether that's sensible? It'd be code that I have to remove later on anyways, std::error::Error does not have a core equivalent unfortunately (I believe it's due to some heap allocation it has to perform) so we couldn't use it and would have to either feature gate it or remove it fully later on.

Reply:
I don't think that no_std/std should be treated as mutually exclusive things.
And don't think that any of std or no_std should have priority in compare with another one.
I use nmea crate for two projects both std: linux/rasberry pi and android phone.

std can be treated as addition to no_std,  we can enable some traits if feature set include std.

But from other hand error with references is very inconvenient, it can not be chained (included into other error to pass to upper layers) without efforts,
so implementation of Display/Error traits are not so important.